src/App.tsx

```diff
@@ -7,18 +7,111 @@ import './App.css'
 import FocusManager from './components/FocusManager'
 
 import AppRoot from './AppRoot'
+import {
+  ScreenReader,
+  AriaScreenReader,
+  SpeechSynthesisTextToSpeech,
+  NullTextToSpeech,
+  TextToSpeech,
+} from './utils/ScreenReader'
 
 /* istanbul ignore next - unsure how to test */
 window.oncontextmenu = (e: MouseEvent): void => {
   e.preventDefault()
 }
 
-const App = () => (
-  <BrowserRouter>
-    <FocusManager>
-      <Route path="/" component={AppRoot} />
-    </FocusManager>
-  </BrowserRouter>
-)
+export interface Props {
+  tts?: {
+    enabled: TextToSpeech
+    disabled: TextToSpeech
+  }
+}
+
+const App = ({
+  tts = {
+    enabled: new SpeechSynthesisTextToSpeech(),
+    disabled: new NullTextToSpeech(),
+  },
+}: Props) => {
+  const [screenReaderEnabled, setScreenReaderEnabled] = useState(false)
+  const [screenReader, setScreenReader] = useState<ScreenReader>(
+    new AriaScreenReader(tts.disabled)
+  )
+
+  /* istanbul ignore next - need to figure out how to test this */
+  const onKeyPress = useCallback(
+    (event: React.KeyboardEvent) => {
+      if (event.key === 'r') {
+        if (screenReaderEnabled) {
+          screenReader.onScreenReaderDisabled()
+          setScreenReader(new AriaScreenReader(tts.disabled))
+          setScreenReaderEnabled(false)
+        } else {
+          const newScreenReader = new AriaScreenReader(tts.enabled)
+          setScreenReader(newScreenReader)
+          setScreenReaderEnabled(true)
+          newScreenReader.onScreenReaderEnabled()
+        }
+      }
+    },
+    [
+      screenReader,
+      setScreenReader,
+      screenReaderEnabled,
+      setScreenReaderEnabled,
+      tts.disabled,
+      tts.enabled,
+    ]
+  )
+
+  /* istanbul ignore next - need to figure out how to test this */
+  const onClick = useCallback(
+    ({ target }: React.MouseEvent) => {
+      if (target) {
+        const currentPath = window.location.pathname
+
+        setImmediate(() => {
+          // Only send `onClick` to the screen reader if the click didn't
+          // trigger navigation.
+          if (window.location.pathname === currentPath) {
+            screenReader.onClick(target)
+          }
+        })
+      }
+    },
+    [screenReader]
+  )
+
+  /* istanbul ignore next - need to figure out how to test this */
+  const onFocus = useCallback(
+    ({ target }: React.FocusEvent) => {
+      if (target) {
+        const currentPath = window.location.pathname
+
+        setImmediate(() => {
```

Comment:
When referencing window methods we've been explicitly using the window method (`window.setTimeout(...)`) in order to clearly identify window methods from local methods.

@eventualbuddha - What do you think about this? Perhaps there is an ESLint rule we can add to enforce this?

Reply:
I don’t feel strongly about it. There already is an eslint rule that enforces things like `window.location` over `location`. Probably just need to expand that.